lib/Config.php

```
@@ -39,6 +39,64 @@ class Config
      */
     private static $environmentConfig;
 
+
+    protected static $systemConfig = null;
+
+    /**
+     * @see    ArrayAccess::offsetExists()
+     *
+     * @param  mixed $offset
+     *
+     * @return bool
+     */
+    public function offsetExists($offset)
+    {
+        return isset(static::$systemConfig[$offset]);
+    }
+
+    /**
+     * @see    ArrayAccess::offsetSet()
+     *
+     * @param  mixed $offset
+     * @param  mixed $value
+     *
+     * @return void
+     */
+    public function offsetSet($offset, $value)
+    {
+        if (is_null($offset)) {
```

I'd throw an Exception here, as modifying the config isn't allowed.